dttools/src/load_average.c

```diff
@@ -45,20 +49,30 @@ void load_average_get(double *avg)
 
 int load_average_get_cpus()
 {
-	FILE *f;
+	struct string_set *cores;
+	cores = string_set_create(0, 0);
+
+	for (int i = 0; ; i++) {
+		char *p = string_format("/sys/devices/system/cpu/cpu%u/topology/thread_siblings", i);
```

Comment:
Tim, could be use /proc/cpuinfo instead and count the number of unique 'core id' lines? This may be a smaller change, since we have been reading /proc/cpuinfo up to now.

Reply:
Oops, I forgot about this PR!

The `core id` lines refer to cores within a physical CPU. On multisocket machines, the `core id`s are not unique across physical CPUs, so that would find the max number of cores in any one CPU.

To use `/proc/cpuinfo`, we would have to parse out multiple fields and count the unique `physical id`, `core id` pairs. That made me nervous since `/proc/cpuinfo` is made for human consumption, so I'm not certain that e.g. field ordering and delimiters are consistent everywhere. So instead, I did the same thing as `lscpu` from `util-linux`.